d2-tracker/dhis2.angular.services.js

```diff
@@ -3702,15 +3690,15 @@ var d2Services = angular.module('d2Services', ['ngResource'])
                             hiddenSections[effect.programStageSection] = effect.programStageSection;
                         }
                     }
-                    else if(effect.action === "SHOWERROR" && effect.dataElement.id){
-                        var headerTxt =  $translate.instant('validation_error');
-                        var bodyTxt = effect.content + (effect.data ? effect.data : "");
-                        NotificationService.showNotifcationDialog(headerTxt, bodyTxt);
-
-                        currentEvent[effect.dataElement.id] = currentEventOriginalValues[effect.dataElement.id];
+                    else if(effect.action === "SHOWERROR" && effect.dataElement && effect.dataElement.id){
+                        if (effect.ineffect) {
+                            errorMessages[effect.dataElement.id] = effect.content + (effect.data ? effect.data : "");
+                        }
                     }
-                    else if(effect.action === "SHOWWARNING"){
-                        warningMessages.push(effect.content + (effect.data ? effect.data : ""));
+                    else if(effect.action === "SHOWWARNING" && effect.dataElement && effect.dataElement.id){
+                        if (effect.ineffect) {
```

Comment:
Remove unneccesary check on .ineffect - already checked in the start of the loop.